File: skyspacefiftyfive_ui/src/services/WeatherAlertService.js
import http from "../http-common"; 

class WeatherAlertService {
  getAllWeatherAlerts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/weatherAlert/weatherAlerts`, searchDTO);
  }

  get(weatherAlertId) {
    return this.getRequest(`/weatherAlert/${weatherAlertId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/weatherAlert?field=${matchData}`, null);
  }

  addWeatherAlert(data) {
    return http.post("/weatherAlert/addWeatherAlert", data);
  }

  update(data) {
  	return http.post("/weatherAlert/updateWeatherAlert", data);
  }
  
  uploadImage(data,weatherAlertId) {
  	return http.postForm("/weatherAlert/uploadImage/"+weatherAlertId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new WeatherAlertService();

File: src/main/java/com/skyspacefiftyfive/service/WeatherAlertService.java
package com.skyspacefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefiftyfive.domain.WeatherAlert;
import com.skyspacefiftyfive.dto.WeatherAlertDTO;
import com.skyspacefiftyfive.dto.WeatherAlertSearchDTO;
import com.skyspacefiftyfive.dto.WeatherAlertPageDTO;
import com.skyspacefiftyfive.dto.WeatherAlertConvertCriteriaDTO;
import com.skyspacefiftyfive.service.GenericService;
import com.skyspacefiftyfive.dto.common.RequestDTO;
import com.skyspacefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface WeatherAlertService extends GenericService<WeatherAlert, Integer> {

	List<WeatherAlert> findAll();

	ResultDTO addWeatherAlert(WeatherAlertDTO weatherAlertDTO, RequestDTO requestDTO);

	ResultDTO updateWeatherAlert(WeatherAlertDTO weatherAlertDTO, RequestDTO requestDTO);

    Page<WeatherAlert> getAllWeatherAlerts(Pageable pageable);

    Page<WeatherAlert> getAllWeatherAlerts(Specification<WeatherAlert> spec, Pageable pageable);

	ResponseEntity<WeatherAlertPageDTO> getWeatherAlerts(WeatherAlertSearchDTO weatherAlertSearchDTO);
	
	List<WeatherAlertDTO> convertWeatherAlertsToWeatherAlertDTOs(List<WeatherAlert> weatherAlerts, WeatherAlertConvertCriteriaDTO convertCriteria);

	WeatherAlertDTO getWeatherAlertDTOById(Integer weatherAlertId);







}





